staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -412,6 +412,15 @@ func (o *ApplyOptions) applyOneObject(info *resource.Info) error {
 		klog.V(4).Infof("error recording current command: %v", err)
 	}
 
+	helper := resource.NewHelper(info.Client, info.Mapping).
+		DryRun(o.DryRunStrategy == cmdutil.DryRunServer)
+
+	if o.DryRunStrategy == cmdutil.DryRunServer {
```

Comment:
Please add a simple comment here about what this is validating (e.g. Ensure the APIServer supports server-side dry run), and maybe what the consequences are (e.g. if we send a mutating call with server-side dry-run, but the APIServer doesn't support it; then it will inadvertently mutate).

Reply:
Commented